ENIGMAsystem/SHELL/Widget_Systems/xlib/dialogs.h

```diff
@@ -0,0 +1,79 @@
+/** Copyright (C) 2019 Samuel Venable
+***
+*** This file is a part of the ENIGMA Development Environment.
+***
+*** ENIGMA is free software: you can redistribute it and/or modify it under the
+*** terms of the GNU General Public License as published by the Free Software
+*** Foundation, version 3 of the license or any later version.
+***
+*** This application and its source code is distributed AS-IS, WITHOUT ANY
+*** WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+*** FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+*** details.
+***
+*** You should have received a copy of the GNU General Public License along
+*** with this code. If not, see <http://www.gnu.org/licenses/>
+**/
+
+#include "Widget_Systems/widgets_mandatory.h"
+#include "Platforms/xlib/XLIBwindow.h"
+#include <string>
+using std::string;
+
+namespace zenity {
```

Comment:
This file gets a lot of simple changes. You'll declare a class, `CommandLineWidgetEngine`, which offers the methods below. Each method will appear like this:
```cpp
namespace enigma {

class CommandLineWidgetEngine {
 public:
  virtual ~CommandLineWidgetEngine() = default;
  virtual int show_message(const string &message) = 0;
  ...
};
```
Notice the `virtual` before and the `= 0` afterward. These just make it so that each system has to implement all methods.

You don't need to declare the Zenity and KDialog methods here. But you do need to declare their containers:

```cpp
extern CommandLineWidgetEngine *zenity_widgets;
extern CommandLineWidgetEngine *kdialog_widgets;

}  // namespace enigma
```